edx_notifications/data.py

```diff
@@ -0,0 +1,169 @@
+"""
+All pure data objects that the various Data Providers will product. This is help avoid
+generic dictionaries from being passed around, plus this will help avoid any
+implicit database-specific bindings that come with any uses of ORMs.
+"""
+
+from django.core.exceptions import ValidationError
+
+from edx_notifications import const
+
+from edx_notifications.base_data import (
+    StringField,
+    DictField,
+    DateTimeField,
+    IntegerField,
+    EnumField,
+    RelatedObjectField,
+    BaseDataObject
+)
+
+
+class NotificationChannel(BaseDataObject):
+    """
+    Specifies a channel through which a notification is delivered
+    """
+
+    # the internal name of the channel
+    name = StringField()
+
+    # the human readible name of the channel
+    display_name = StringField()
+
+    # the human readible description of the channel
+    description_name = StringField()
+
+
+class NotificationType(BaseDataObject):
+    """
+    The Data Object representing the NotificationType
+    """
+
+    # the name (including namespace) of the notification, e.g. open-edx.lms.forums.reply-to-post
+    name = StringField()
+
+    # the human readible string of the name of the notification
+    display_name = StringField()
+
+    # the human readible string that describes the notification
+    display_description = StringField()
+
+    # default delivery channel for this type
+    # None = no default
+    default_channel = RelatedObjectField(NotificationChannel)
+
+    # renderer class - as a string - that will handle the rendering of
+    # this type
+    renderer = StringField()
```

Comment:
Isn't the `renderer` a property of the channel, not of the notification type? The format of the output is going to depend heavily on how the message is getting to the user, right?